base/dcp_receiver.go

```diff
@@ -202,9 +204,90 @@ type NoPasswordAuthHandler struct {
 
 func (nph NoPasswordAuthHandler) GetCredentials() (username string, password string, bucketname string) {
 	_, _, bucketname = nph.Handler.GetCredentials()
+
 	return "", "", bucketname
 }
 
+// getHighSeqMetadata returns metadata to feed into a DCP client based on the last sequence numbers stored in memory
+func getHighSeqMetadata(bucket Bucket) ([]DCPMetadata, error) {
+	store, ok := AsCouchbaseStore(bucket)
+	if !ok {
+		return nil, errors.New("DCP Client requires bucket to be CouchbaseStore")
+	}
+
+	numVbuckets, err := store.GetMaxVbno()
+	if err != nil {
+		return nil, fmt.Errorf("Unable to determine maxVbNo when creating DCP client: %w", err)
+	}
+
+	_, highSeqNos, statsErr := store.GetStatsVbSeqno(numVbuckets, true)
+	if statsErr != nil {
+		return nil, fmt.Errorf("Unable to obtain high seqnos for one-shot DCP feed: %w", statsErr)
+	}
+
+	metadata := make([]DCPMetadata, numVbuckets)
+	for vbNo := uint16(0); vbNo < numVbuckets; vbNo++ {
+		// TODO: should we/can we initialize VbUUID or other fields?
```

Comment:
I think so. Should be able to use the slice returned by `GetSTatsVbSeqno` with this index - although I have no idea about the failover entries...

Reply:
This is probably a question for @adamcfraser - these will be populated when the DCPClient starts, is my understanding.